Feature/public access filter

gmag11 opened the pull request:
This introduces a public access filter system, enabling fine-grained control over which accounts, asset classes, holdings, and tags are accessible to public or private users.
This update allows users to define exactly which resources are accessible when sharing access, improving security and flexibility for public scenario.

gmag11 committed:
Task/add filter functionality to access dialog component

gmag11 committed:
Task/refactor access dialog to enhance filter settings and improve form structure

gmag11 committed:
Task/add access filter and settings interfaces, update access DTOs and components for filter functionality

gmag11 committed:
Task/add comprehensive filter functionality to public portfolio retrieval

gmag11 committed:
Task/update portfolio and activity filters to handle multiple holdings and improve filtering logic

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter

gmag11 committed:
Refactor: Reorder methods

gmag11 committed:
Refactor: Improve error handling and form validation in access dialogs

gmag11 committed:
Add public access filtering feature to changelog

dtslvr commented on apps/api/src/app/access/access-filter.dto.ts:
```diff
@@ -0,0 +1,40 @@
+  ValidateNested
+} from 'class-validator';
+
+export class HoldingFilterDto {
```
No export needed

dtslvr commented on apps/api/src/app/access/access-filter.dto.ts:
```diff
@@ -0,0 +1,40 @@
+  @IsArray()
+  @IsOptional()
+  @ValidateNested({ each: true })
+  @Type(() => HoldingFilterDto)
```
Move up ⬆️

dtslvr commented on apps/api/src/app/access/access.controller.ts:
```diff
@@ -122,6 +94,48 @@ export class AccessController {
     });
   }
 
+  @Get()
```
Move up ⬆️

gmag11 replied:
I've done that

dtslvr commented on apps/api/src/app/access/access.controller.ts:
```diff
@@ -122,6 +94,48 @@ export class AccessController {
+          id,
+          permissions: accessPermissions,
+          settings: settings as AccessSettings,
+          grantee: 'Public',
```
Move up ⬆️

dtslvr commented on apps/api/src/app/endpoints/public/public.controller.ts:
```diff
@@ -61,43 +65,155 @@ export class PublicController {
+
+    if (accessFilter) {
+      // Add account filters
+      if (accessFilter.accountIds && accessFilter.accountIds.length > 0) {
```
`if (accessFilter.accountIds?.length > 0) {`

dtslvr commented on apps/api/src/app/endpoints/public/public.controller.ts:
```diff
@@ -61,43 +65,155 @@ export class PublicController {
       })
     ]);
 
+    // Filter out only the base currency cash holdings
```
❓

dtslvr commented on apps/api/src/app/endpoints/public/public.controller.ts:
```diff
@@ -61,43 +65,155 @@ export class PublicController {
       sortColumn: 'date',
       sortDirection: 'desc',
-      take: 10,
+      take: hasMultipleHoldingFilters ? 1000 : 10, // Get more if we need to filter manually
```
❓

gmag11 replied:
I did not understood how filtering was passed to orderService. So I did a quick test to try different things. This is a leftover. I removed it and fixed another issue that I had below.

> ❓

dtslvr commented on apps/api/src/app/endpoints/public/public.controller.ts:
```diff
@@ -61,43 +65,155 @@ export class PublicController {
       withExcludedAccountsAndActivities: false
     });
 
+    // If multiple holdings, filter activities manually
```
❓

dtslvr commented on apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.html:
```diff
@@ -59,6 +59,80 @@ <h1 mat-dialog-title>
+
+    @if (showFilterPanel) {
+      <div class="mt-4">
+        <h2 class="mb-3" i18n style="font-size: 1.125rem; font-weight: 500">
```
Don't use inline styles

dtslvr commented on apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.html:
```diff
@@ -59,6 +59,80 @@ <h1 mat-dialog-title>
+          visible in this public access. Leave empty to show all data.
+        </p>
+
+        <div class="mb-3">
```
Why don't you reuse the component from #5605 you have created exactly for this use case? 😱

dtslvr reviewed (changes requested):
Thanks for your contribution, @gmag11!
I have reviewed your changes and added a few comments.

Could you also please remove the Spanish comments? Our project standard requires comments to be in English to ensure consistency and maintainability across all collaborators. Thank you!

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter

gmag11 committed:
Feature: Refactor access filter dialog and improve filter handling

gmag11 committed:
Feature: Refactor holding filters in public portfolio retrieval

gmag11 committed:
Feature: Rearrange properties in public portfolio response structure

gmag11 committed:
Feature: Update Access interfaces to use AssetProfileIdentifier for holdings

gmag11 committed:
Feature: Remove comments in access dialog component for clarity

gmag11 committed:
Feature: Reorganize create and delete access methods in AccessController for improved clarity

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter

dtslvr commented on apps/api/src/app/access/access-filter.dto.ts:
```diff
@@ -0,0 +1,40 @@
+  ValidateNested
+} from 'class-validator';
+
+class HoldingFilterDto {
```
Could you rename to `AssetProfileIdentifierDto` (inspired by `AssetProfileIdentifier` in `libs/common/src/lib/interfaces`) and move to `libs/common/src/lib/dto`

dtslvr commented on apps/api/src/app/access/access.controller.ts:
```diff
@@ -169,4 +166,25 @@ export class AccessController {
     }
   }
+
+  @Delete(':id')
```
Why has this been moved? Moving code like this makes the review diff much harder to follow.

dtslvr commented on apps/api/src/app/access/create-access.dto.ts:
```diff
@@ -6,6 +8,9 @@ export class CreateAccessDto {
   @IsString()
   alias?: string;
 
+  @IsOptional()
```
`@IsObject()` ❓

gmag11 replied:
Should `@isOptional()` be removed?. In this case form is not accepted if there is no filter configured. I add `@IsObject()` decorator before `@isOptional()`

dtslvr commented on apps/api/src/app/access/update-access.dto.ts:
```diff
@@ -6,6 +8,9 @@ export class UpdateAccessDto {
   @IsString()
   alias?: string;
 
+  @IsOptional()
```
`@IsObject()` ❓

gmag11 replied:
Same as `create-access.dto.ts`

dtslvr commented on apps/api/src/app/endpoints/public/public.controller.ts:
```diff
@@ -61,27 +65,123 @@ export class PublicController {
     }
 
+    // Get filter configuration from access settings
+    const accessSettings = (access.settings ?? {}) as AccessSettings;
```
`const { filter } = ...` ❓

dtslvr commented on apps/api/src/app/endpoints/public/public.controller.ts:
```diff
@@ -61,27 +65,123 @@ export class PublicController {
+        );
+      }
+
+      // Add holding filters (symbol + dataSource)
```
Move up ⬆️ (after asset class)

dtslvr commented on apps/api/src/app/endpoints/public/public.controller.ts:
```diff
@@ -61,27 +65,123 @@ export class PublicController {
       })
     ]);
 
+    const baseCurrency =
```
Why is it not enough to use filters?

dtslvr commented on apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts:
```diff
@@ -119,11 +155,164 @@ export class GfCreateOrUpdateAccessDialogComponent
+  private loadFilterData() {
+    const existingFilter = this.data.access.settings?.filter;
+
+    this.dataService
```
The `user` object (via `userService`) contains the `accounts`, no need to fetch here

dtslvr reviewed (changes requested):
Thanks for the update, @gmag11.
I have reviewed your changes and added a few comments.

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter

gmag11 committed:
feat: add AssetProfileIdentifierDto and update access DTOs to use it

gmag11 commented on apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts:
```diff
@@ -89,17 +114,28 @@ export class GfCreateOrUpdateAccessDialogComponent
       this.changeDetectorRef.markForCheck();
     });
+
+    if (isPublic) {
```
None. It is defined as `const isPublic = this.data.access.type === 'PUBLIC';` as it is used a couple times.

`accessType` is not accessible in that context as it is the value of access type when user fill the form in.

gmag11 commented on apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts:
```diff
@@ -119,11 +155,164 @@ export class GfCreateOrUpdateAccessDialogComponent
+        this.changeDetectorRef.markForCheck();
+      });
+
+    this.dataService
```
I've got accounts and tags from user. But it is still needed to fetch holdings and asset classes, isn't it?

gmag11 committed:
refactor: streamline access filter handling in PublicController and update access dialog to use UserService

gmag11 committed:
refactor: enhance filter settings layout and styling in access dialog

gmag11 commented:
> Thanks for the update, @gmag11. I have reviewed your changes and added a few comments.

Thanks @dtslvr . I've reviewed and implemented proposed changes but I need a couple clarifications. Check unresolved issues here.

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter

gmag11 committed:
Refactor DTO imports and restructure asset profile identifier DTO

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter

gmag11 committed:
Changelog: add filtering options for public access by account, holdings, tag, or asset class

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter

gmag11 committed:
Update changelog

gmag11 committed:
Merge branch 'main' into feature/public-access-filter

gmag11 committed:
Merge commit '0be83cf033558bc67d4d498e425773b2dd484e6d' into feature/public-access-filter

# Conflicts:
#	apps/api/src/app/endpoints/public/public.controller.ts
#	apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts
#	apps/client/src/app/components/user-account-access/user-account-access.component.ts

gmag11 committed:
Merge commit 'f14f98f766d70dd6c9b5e2c13cf0d67ab927a33a' into feature/public-access-filter

gmag11 committed:
Merge remote-tracking branch 'upstream/main' into feature/public-access-filter
